plugins/firehydrant/src/components/ServiceDetailsCard/ServiceDetailsCard.tsx

```diff
@@ -0,0 +1,285 @@
+      error={analyticsError}
+    />
+  );
+};
+
+export const ServiceDetailsCard = () => {
+  const { entity } = useEntity();
+  const classes = useStyles();
+  const [showServiceDetails, setShowServiceDetails] = useState(false);
+  const configApi = useApi(configApiRef);
+
+  const BASE_URL =
+    configApi.getOptionalString('firehydrant.baseUrl') ||
+    'https://app.firehydrant.io';
+
+  const startDate = moment().subtract(30, 'days').utc();
+  const endDate = moment().utc();
+
+  // The Backstage service name in FireHydrant is a unique formatted string
+  // that requires the entity's kind, name, and namespace.
+  const fireHydrantServiceName = `${entity?.kind}:${
+    entity?.metadata?.namespace ?? 'default'
+  }/${entity?.metadata?.name}`;
+
+  const { loading, value, error } = useServiceDetails({
+    serviceName: fireHydrantServiceName,
+  });
+
+  const activeIncidents: string[] = value?.service?.active_incidents ?? [];
+  const incidents: ServiceIncidentsResponse = value?.incidents ?? [];
+  const serviceId: string = value?.service?.id!;
+
+  useEffect(() => {
+    if (value?.service && Object.keys(value?.service).length > 0) {
+      setShowServiceDetails(true);
+    }
+  }, [value]);
+
+  if (loading) {
+    return <Progress />;
+  }
+
+  if (error) {
+    return (
+      <ServiceWarning text="There was an issue connecting to FireHydrant." />
+    );
+  }
+
+  const headerText: string = showServiceDetails
+    ? `There ${activeIncidents?.length === 1 ? 'is' : 'are'} ${
+        activeIncidents?.length
+      } active incident${activeIncidents?.length === 1 ? '' : 's'}.`
+    : '';
+
+  const serviceIncidentsLink: string = `${BASE_URL}/incidents?search={"services":[{"label":${JSON.stringify(
+    value?.service?.name,
+  )},"value":${JSON.stringify(value?.service?.id)}}]}`;
+
+  return (
+    <InfoCard className="infoCard">
+      {!showServiceDetails && !loading && (
+        <ServiceWarning text="This service does not exist in FireHydrant." />
+      )}
+      {showServiceDetails && (
+        <Box
+          alignItems="center"
+          display="flex"
+          justifyContent="space-between"
+          borderBottom="1px solid #d5d5d5"
+        >
+          <Box>
+            <h2>{headerText}</h2>
+          </Box>
+          <Box>
+            <MaterialButton
+              className={classes.buttonLink}
+              color="default"
+              href={serviceIncidentsLink}
+              target="_blank"
+              variant="outlined"
+            >
+              View service incidents&nbsp;&nbsp;
+              <ExitToAppIcon />
+            </MaterialButton>
+          </Box>
+        </Box>
+      )}
+      {activeIncidents && activeIncidents?.length > 0 && (
+        <Box className={classes.linksContainer}>
+          {incidents &&
+            incidents?.slice(0, 5).map((incident: Incident, index: number) => (
+              <div key={index}>
+                <a className={classes.link} href={incident.incident_url}>
```

Comment:
Consider using the builtin `Link` that we have in `core-components` over `a` at all times (here and below). It is the one that will always behave correctly with react-router, have click tracking / metrics if enabled, sets target/rel depending on the right set of rules etc.